### PR TITLE
Pouchdb sync

### PR DESCRIPTION
**Pull request by stevekroh**
Make a local PouchDB replicate from the DB server.
Also, begin integration of PouchDB into the redux store.

### CODE REVIEW

**Comment by ethanjurman** [comment]
:facepunch: :camel:

File: www/js/app.js [hunk 1]
```diff
@@ -2,14 +2,13 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 
 // redux components
-import { createStore } from 'redux';
 import { Provider } from 'react-redux';
-import clientApp from './reducers/reducer';
+// import clientApp from './reducers/reducer';
```
Comment (JRJurman) [inline www/js/app.js]: probably should remove this?

File: www/js/store.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+export function createAppStore(initState) {
+  return finalCreateStore(app, initState);
+}
+
+// const remoteDb = new PouchDB('http://52.21.125.160:5984/points');
+
+// const rep = PouchDB.replicate(remoteDb, localDb).on('complete', info => {
+//
+// });
```
Comment (ethanjurman) [inline www/js/store.js]: remove commented code?

File: www/js/app.js [hunk 3]
```diff
@@ -20,8 +19,9 @@ import DownloadTrackPage from './components/download-track-page';
 import FilterPage from './components/filter-page';
 import SettingsPage from './components/settings-page';
 
-import rest from 'rest';
-import mime from 'rest/interceptor/mime';
+import {loadDb} from './db'
+import {resetPoints} from './actions/point-actions';
```
Comment (JRJurman) [inline www/js/app.js]: is this ever called?

File: www/js/components/add-point-page.js [hunk 4]
```diff
@@ -51,7 +51,7 @@ export class AddPointPage extends Component {
           <Marker position={this.state.center} radius={10}/>
         </Map>
 
-        <AddPointCard latlng={this.state.center} />
+        <AddPointCard {...this.props } latlng={this.state.center} />
```
Comment (JRJurman) [inline www/js/components/add-point-page.js]: what props do we actually need to pass down?

File: www/js/reducers/points.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+
+import {ADD_POINT, RESET_POINTS} from '../actions/point-actions'
+
+export function points(state = [], action) {
+  switch(action.type) {
+    case ADD_POINT:
+      return [
+        ...state,
+        Object.assign({}, action.point)
```
Comment (JRJurman) [inline www/js/reducers/points.js]: what is this `Object.assign()`

File: www/js/store.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+const finalCreateStore = compose(
+  applyMiddleware(persister),
+  window.devToolsExtension ? window.devToolsExtension() : f => f
+)(createStore);
+
+export function createAppStore(initState) {
+  return finalCreateStore(app, initState);
+}
+
+// const remoteDb = new PouchDB('http://52.21.125.160:5984/points');
```
Comment (JRJurman) [inline www/js/store.js]: should this be removed?

**Comment by JRJurman** [comment]
do we still need `www/js/ac-pouch.js`

File: www/js/reducers/points.js (hunk 5 above)
Comment (stevekroh) [inline www/js/reducers/points.js]: `Object.assign()` makes copies of objects. However we actually don't need it here. I think we only need to make copies of objects originating from the state argument.

File: www/js/store.js (hunk 6 above)
Comment (stevekroh) [inline www/js/store.js]: Nope, this will be refactored eventually, but we need the pouch middleware and redux extension helper.

**Comment by stevekroh** [comment]
I will get rid of `www/js/ac-pouch.js` now since it is obsolete after this PR

**Comment by JRJurman** [comment]
:tada: :cactus:

**Pull request merged**